azimuth/routers/v1/utterances.py

```diff
@@ -264,6 +265,47 @@ def get_utterances(
     )
 
 
+@router.post(
+    "",
+    summary="Patch utterances",
+    description="Patch utterances, such as updating proposed actions.",
+    tags=TAGS,
+    response_model=List[UtterancePatch],
+)
+def patch_utterances(
+    request_data: List[UtterancePatch] = Body(...),
+    dataset_split_manager: DatasetSplitManager = Depends(get_dataset_split_manager),
+    task_manager: TaskManager = Depends(get_task_manager),
+) -> List[UtterancePatch]:
+    persistent_ids = [utt.persistent_id for utt in request_data]
+    try:
+        row_indices = dataset_split_manager.get_row_indices_from_persistent_id(persistent_ids)
+    except ValueError as e:
+        raise HTTPException(HTTP_404_NOT_FOUND, detail=f"Persistent id not found: {e}.")
+
+    request_data_complete = {}
+    for row_idx, utterance in zip(row_indices, request_data):
+        request_data_complete[row_idx] = {data_action: False for data_action in ALL_DATA_ACTIONS}
+        if utterance.data_action != DataAction.no_action:
+            request_data_complete[row_idx][utterance.data_action] = True
+
+    dataset_split_manager.add_tags(request_data_complete)
+
+    task_manager.clear_worker_cache()
+    updated_tags = dataset_split_manager.get_tags(row_indices)
+
+    return [
+        UtterancePatch(
+            persistent_id=persistent_id,
+            data_action=next(
+                iter(tag for tag, value in tags.items() if tag in ALL_DATA_ACTIONS and value),
```

Comment:
I just checked and we don't need to create an iterator; the generator should suffice. I'm not sure why we have `next(iter(...)` everywhere in the repo. Because we supply the default value too, we need some parentheses around the generator, though, otherwise we get a syntax error. Maybe that's what confused us into using `iter()` before. 🤷‍♂️ 
```suggestion
               (tag for tag, value in tags.items() if tag in ALL_DATA_ACTIONS and value),
```

Reply:
Ah! The `iter()` is necessary when we use a list or some `dict.values()`, which are not iterators (whereas generators are). I cleaned the ones I could in https://github.com/ServiceNow/azimuth/pull/429/commits/1379630459d185a444558543c5513320c418d4cb